src/redisearch_rs/rqe_iterators_bencher/src/ffi.rs

```diff
@@ -74,7 +76,19 @@ impl QueryIterator {
         }
         Self(unsafe { bindings::NewIdListIterator(data, len as u64, 1f64) })
     }
-
+    #[inline(always)]
+    pub fn new_metric(vec: Vec<u64>, metric_data: Vec<f64>) -> Self {
+        let len = vec.len();
+        let data =
+            unsafe { RedisModule_Alloc.unwrap()(len * std::mem::size_of::<u64>()) as *mut u64 };
```

Comment:
We can use the Rust allocation API here.

We call `redis_mock::bind_redis_alloc_symbols_to_mock_impl!();
`  in `lib.rs` which will ensure the right allocator is used.

```rust
        let layout = std::alloc::Layout::array::<u64>(len).unwrap();
        let data = unsafe { std::alloc::alloc(layout) } as *mut u64;
        let m_data = unsafe { std::alloc::alloc(layout) } as *mut f64;
```